src/main/payment/main/G1HRPS/PaymentManager.java

```diff
@@ -15,32 +15,194 @@
+	 * @param status
+	 */
+
+	public void AddNewObject(UUID id, String guest_id, int room_num, float discounts, float tax, float bill_total, PaymentStatus status) {
+
+		Payment new_payment = new Payment(id, guest_id, room_num, discounts, tax, bill_total, status);
+
+		AddToList(new_payment);
+
 	}
 
 	/**
 	 * Takes in an class object and list to add the object into.
+	 * 
+	 * @param payment
 	 */
 	public void AddToList(Payment payment) {
-		payment_list_.add(payment);
+
+		try{
+			payment_list_.add(payment);
+		}
+		catch(NullPointerException e){
+			System.out.println("Payment List not initialized");
+			e.printStackTrace();
+		}
+
 	}
 
 	/**
 	 * Takes in an class object and list to remove the object from the given list.
+	 * 
+	 * @param payment
 	 */
 	public void RemoveFromList(Payment payment) {
-		// TODO - implement PaymentManager.RemoveFromList
-		throw new UnsupportedOperationException();
+
+		try{
+			payment_list_.remove(payment);
+		}
+		catch(NullPointerException e){
+			System.out.println("Payment List not initialized");
+			e.printStackTrace();
+		}
+
+	}
+
+	/**
+	 * Method overrides the <T> SearchList(String search_text)
+	 * in Supermanager interface and returns a payment with
+	 * matching search text
+	 * 
+	 * @param 	payment_id	parameter is given as String and
+	 * 						is used to get the corresponding
+	 * 						UUID of payment
+	 * 
+	 * @return	payment with the matching payment_id
+	 */
+
+	@Override
+	public Payment SearchList(String key) {
+
+		UUID uuid = UUID.fromString(key);
+
+		for(Payment payment : payment_list_){
+			if(uuid.equals(payment.GetPaymentID())){
+				return payment;
+			}
+		}
+
+		return null;
+
 	}
 
-	public void SearchList(String search_text) {
-		// TODO - implement PaymentManager.SearchList
-		throw new UnsupportedOperationException();
+	/**
+	 * 
+	 * @param room_num
+	 * @return pending payment on the room with matching room number
+	 */
+
+	public Payment SearchList(int room_num) {
+
+		for(Payment payment : payment_list_){
+			if((room_num == payment.GetRoomNum()) && 
+				(payment.GetStatus() == PaymentStatus.Pending)){
+				return payment;
+			}
+		}
+
+		return null;
+
 	}
 
+	/**
+	 * 
+	 * @return the list of all payments
+	 */
+
 	public List<Payment> GetList() {
 		return payment_list_;
 	}
 
+	/**
+	 * Takes new payment status and assigns it to the given payment
+	 * 
+	 * @param payment
+	 * @param status
+	 */
+	public void SetPaymentStatus(Payment payment, PaymentStatus status) {
+
+		payment.SetStatus(status);
+
+	}
+
+	/**
+	 *
+	 * @param payment
+	 * @return	PaymentStatus of the given payment
+	 */
+
+	public PaymentStatus GetPaymentStatus(Payment payment) {
+
+		return payment.GetStatus();
+
+	}
+
+	/**
+	 * Get linked items from room service and room charges and print the list
+	 * 
+	 * @param payment
+	 * @param days_of_stay
+	 * @param rs_order_list
+	 */
+	public void GenerateAndPrintBill(Payment payment, int days_of_stay, List<RoomServiceOrder> rs_order_list) {
```

Comment:
GenerateAndPrintBill should be the one that creates the Payment object.

Reply:
I'll just rename the method to `GenerateBill()` and override `toString()` for Payment class.